src/main/java/fi/aalto/cs/apluscourses/intellij/utils/CourseFileManager.java

```diff
@@ -216,5 +304,29 @@ private void loadFromJsonObject(@NotNull JSONObject jsonObject) throws IOExcepti
       this.modulesMetadata
           .put(moduleName, new ModuleMetadata(moduleVersion, downloadedAt));
     }
+
+
+    JSONObject dependenciesObject = jsonObject.optJSONObject(MODULE_DEPENDENCIES_KEY);
+    if (dependenciesObject != null) {
+      Iterable<String> depModuleNames = dependenciesObject::keys;
+
+      for (String moduleName : depModuleNames) {
+        Set<String> dependencies = new HashSet<>();
+        JSONArray jsonDependencies = dependenciesObject.getJSONArray(moduleName);
+        for (int i = 0; i < jsonDependencies.length(); i++) {
+          dependencies.add(jsonDependencies.getString(i));
+        }
+
```

Comment:
Please, put this logic in a utility function. You could write:

```java
dependencies = new HashSet<>(Arrays.asList(JsonUtil.parseArray(jsonDependencies, JSONArray::getString, Function.identity(), String[]::new)))
```

but that's kind of ugly, so perhaps adding new utility function(s) and/or modifying the existing ones would be advisable.

Reply:
I think it's beautiful, so I'll use it